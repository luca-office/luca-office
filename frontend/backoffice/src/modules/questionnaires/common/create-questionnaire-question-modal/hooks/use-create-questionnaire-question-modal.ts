import * as React from "react"
import {QuestionScoringType, QuestionType} from "shared/graphql/generated/globalTypes"
import {useCreateQuestionnaireQuestion} from "shared/graphql/hooks"
import {Option} from "shared/utils"

export interface UseCreateQuestionnaireQuestionModalHook {
  readonly createQuestion: (questionType: QuestionType) => void
  readonly selectedQuestionType: Option<QuestionType>
  readonly setSelectedQuestionType: (questionType: QuestionType) => void
}

export const useCreateQuestionnaireQuestionModal = (
  questionnaireId: UUID,
  navigateToQuestion: (questionId: UUID) => void
): UseCreateQuestionnaireQuestionModalHook => {
  const [selectedQuestionType, setSelectedQuestionType] = React.useState<Option<QuestionType>>(Option.none())

  const {createQuestionnaireQuestion} = useCreateQuestionnaireQuestion()

  const handleSetSelectedQuestionType = (questionType: QuestionType) => setSelectedQuestionType(Option.of(questionType))

  const createQuestion = (questionType: QuestionType) => {
    createQuestionnaireQuestion({
      text: "",
      questionType,
      isAdditionalFreeTextAnswerEnabled: false,
      questionnaireId,
      scoringType: QuestionScoringType.None,
      score: selectedQuestionType.map(type => (type === QuestionType.SingleChoice ? 1 : 0)).getOrElse(0)
    }).then(result => {
      result.forEach(question => {
        navigateToQuestion(question.id)
      })
    })
  }

  return {createQuestion, selectedQuestionType, setSelectedQuestionType: handleSetSelectedQuestionType}
}
