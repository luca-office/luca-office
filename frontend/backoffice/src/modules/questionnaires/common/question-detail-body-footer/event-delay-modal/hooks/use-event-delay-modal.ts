import isEmpty from "lodash-es/isEmpty"
import {useForm, UseFormMethods} from "react-hook-form"
import {SelectOptionCustomized} from "shared/components"
import {TimeUnit} from "shared/enums"
import {useLucaTranslation} from "shared/translations"
import {convertSecondsToMinutes} from "shared/utils"

export interface EventDelayForm {
  readonly delay: string
  readonly delayUnit: string
}

export interface UseEventDelayModalHook {
  readonly formMethods: UseFormMethods<EventDelayForm>
  readonly isConfirmDisabled: boolean
  readonly unitOptions: SelectOptionCustomized[]
  readonly errorMessage?: string
}

export const useEventDelayModal = (delay?: number): UseEventDelayModalHook => {
  const {t} = useLucaTranslation()

  const formMethods = useForm<EventDelayForm>({
    defaultValues: {delay: delay ? `${convertSecondsToMinutes(delay)}` : "", delayUnit: TimeUnit.Minute}
  })

  const unitOptions = [{label: t("unit__minutes_plural"), value: TimeUnit.Minute}] // only minutes apply here
  const isConfirmDisabled = !isEmpty(formMethods.errors)

  return {
    formMethods,
    isConfirmDisabled,
    unitOptions
  }
}
