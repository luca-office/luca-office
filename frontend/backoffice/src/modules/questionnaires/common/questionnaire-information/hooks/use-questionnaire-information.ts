import {useState} from "react"
import {useDispatch} from "react-redux"
import {ColumnProps} from "shared/components"
import {QuestionnaireUpdate} from "shared/graphql/generated/globalTypes"
import {NavigationConfig, Questionnaire, QuestionnaireQuestion} from "shared/models"
import {useLucaTranslation} from "shared/translations"
import {isDefined} from "shared/utils"
import {OverlayEditCompositeFieldConfig, OverlayEditFieldType} from "../../../../../components"
import {navigateToRouteAction} from "../../../../../redux/actions/navigation-action"
import {Route} from "../../../../../routes"
import {getQuestionsColumns} from "../../../config"

export interface UseQuestionnaireInformationHook {
  readonly descriptionField: OverlayEditCompositeFieldConfig
  readonly isEditable: boolean
  readonly isEditDurationModalVisible: boolean
  readonly navigateToQuestionDetails: (questionId: UUID) => void
  readonly questionnaireUpdate: QuestionnaireUpdate
  readonly questionsColumns: ColumnProps<QuestionnaireQuestion>[]
  readonly toggleIsEditDurationModalVisible: () => void
}

export type UseQuestionnaireInformationParams = {
  readonly deleteQuestion: (id: UUID) => void
  readonly hideActions: boolean
  readonly navigationQuestionConfig: NavigationConfig<Route>
  readonly openQuestionCreation: () => void
  readonly openQuestionsSortOverlay: () => void
  readonly questionnaire: Questionnaire
  readonly readonly: boolean
  readonly showQuestionScoring: boolean
}

export const useQuestionnaireInformation = ({
  deleteQuestion,
  hideActions,
  navigationQuestionConfig,
  openQuestionCreation,
  openQuestionsSortOverlay,
  questionnaire,
  readonly,
  showQuestionScoring
}: UseQuestionnaireInformationParams): UseQuestionnaireInformationHook => {
  const {t} = useLucaTranslation()

  const dispatch = useDispatch()

  const [isEditDurationModalVisible, setIsEditDurationModalVisible] = useState(false)

  const isEditable = !readonly && !isDefined(questionnaire.finalizedAt) && !isDefined(questionnaire.publishedAt)

  const descriptionField: OverlayEditCompositeFieldConfig = {
    updateId: "description",
    value: questionnaire.description,
    labelKey: "description",
    type: OverlayEditFieldType.TEXTAREA
  }

  const questionnaireUpdate: QuestionnaireUpdate = {
    binaryFileId: questionnaire.binaryFileId,
    description: questionnaire.description,
    questionnaireType: questionnaire.questionnaireType,
    title: questionnaire.title
  }

  const questionsColumns = getQuestionsColumns({
    openSortOverlay: openQuestionsSortOverlay,
    openCreation: openQuestionCreation,
    deleteQuestion,
    showQuestionScoring,
    isEditable,
    hideActionButtons: hideActions,
    t
  })

  const navigateToQuestionDetails = (questionId: UUID) => {
    dispatch(navigateToRouteAction(navigationQuestionConfig.route, {...navigationQuestionConfig.payload, questionId}))
  }

  return {
    isEditable,
    descriptionField,
    questionnaireUpdate,
    questionsColumns,
    navigateToQuestionDetails,
    isEditDurationModalVisible,
    toggleIsEditDurationModalVisible: () => setIsEditDurationModalVisible(!isEditDurationModalVisible)
  }
}
