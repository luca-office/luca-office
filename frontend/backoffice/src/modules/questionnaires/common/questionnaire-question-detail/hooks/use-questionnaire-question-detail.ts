import {pick} from "lodash-es"
import * as React from "react"
import {useDispatch} from "react-redux"
import {InterventionGroupType, InterventionHeaderGroupType} from "shared/enums"
import {
  QuestionnaireAnswerUpdate,
  QuestionnaireQuestionUpdate,
  QuestionType
} from "shared/graphql/generated/globalTypes"
import {
  useCreateFreetextQuestionCodingCriterion,
  useCreateQuestionnaireAnswer,
  useDeleteFreetextQuestionCodingCriterion,
  useDeleteQuestionnaireAnswer,
  useDeleteQuestionnaireQuestion,
  useInterventions,
  useQuestionnaireQuestion,
  useRepositionQuestionnaireAnswer,
  useScenario,
  useUpdateFreetextQuestionCodingCriterion,
  useUpdateQuestionnaireAnswer,
  useUpdateQuestionnaireQuestion
} from "shared/graphql/hooks"
import {
  FreetextQuestionCodingCriterion,
  QuestionnaireAnswer,
  QuestionnaireQuestion,
  RuntimeSurveyAnswerSelectionIntervention
} from "shared/models"
import {Option} from "shared/utils"
import {ResortableEntity, ResortedEntity} from "../../../../../models"
import {navigateToRouteAction} from "../../../../../redux/actions/navigation-action"
import {Route} from "../../../../../routes"
import {getInterventionsForTypeName} from "../../../../scenario-interventions/utils"
import {QuestionAnswerTableEntity} from "../../../config"
import {convertAnswersToResortableEntity} from "../../../utils"

interface QuestionnaireQuestionInterventionsConfig {
  readonly isCreateInterventionModalVisible: boolean
  readonly scenarioMaxDurationInSeconds: number
  readonly runtimeSurveyAnswerSelectionInterventions: RuntimeSurveyAnswerSelectionIntervention[]
  readonly setSelectedAnswerForInterventionsSetting: React.Dispatch<
    React.SetStateAction<Option<QuestionAnswerTableEntity>>
  >
  readonly toggleIsCreateInterventionModalVisible: () => void
  readonly selectedAnswerForInterventionsSetting: Option<QuestionAnswerTableEntity>
  readonly navigateToInterventions: () => void
  readonly isScenarioReadOnly: boolean
}

export interface UseQuestionnaireQuestionDetailHook {
  readonly question: Option<QuestionnaireQuestion>
  readonly actionsLoading: boolean
  readonly dataLoading: boolean
  readonly deleteAnswer: (id: UUID) => void
  readonly deleteQuestion: (id: UUID) => Promise<void>
  readonly handleCreateAnswer: () => void
  readonly isChangeQuestionTypeModalVisible: boolean
  readonly interventionsConfig: QuestionnaireQuestionInterventionsConfig
  readonly openAnswersSortOverlay: () => void
  readonly repositionAnswers: (orderedEntities: ResortedEntity[]) => Promise<Option<QuestionnaireAnswer>[]>
  readonly resortModalVisible: boolean
  readonly setChangeQuestionTypeModalVisible: (modalVisible: boolean) => void
  readonly setResortModalVisible: (isVisible: boolean) => void
  readonly sortableAnswers: Option<ResortableEntity[]>
  readonly toggleFreeTextAnswer: (isAdditionalTextAnswerAllowed: boolean) => void
  readonly updateQuestionnaireQuestion: (update: QuestionnaireQuestionUpdate) => void
  readonly updateQuestionnaireAnswer: (
    answerId: UUID,
    update: QuestionAnswerTableEntity
  ) => Promise<Option<QuestionnaireAnswer>> | Promise<Option<FreetextQuestionCodingCriterion>>
  readonly createDefaultHolisticAnswer: () => void
}

export const useQuestionnaireQuestionDetail = (
  questionnaireId: UUID,
  questionId: UUID,
  navigateToQuestionnaire: () => void,
  scenarioIdForInterventions?: UUID // Use only for interventions
): UseQuestionnaireQuestionDetailHook => {
  const [isChangeQuestionTypeModalVisible, setChangeQuestionTypeModalVisible] = React.useState(false)
  const [resortModalVisible, setResortModalVisible] = React.useState(false)
  const [selectedAnswerForInterventionsSetting, setSelectedAnswerForInterventionsSetting] = React.useState<
    Option<QuestionAnswerTableEntity>
  >(Option.none())
  const [isCreateInterventionModalVisible, setIsCreateInterventionModalVisible] = React.useState(false)

  const dispatch = useDispatch()

  const {questionnaireQuestion: questionOption, questionnaireQuestionLoading} = useQuestionnaireQuestion(questionId)
  const {interventions, interventionsLoading} = useInterventions(
    scenarioIdForInterventions ?? "",
    scenarioIdForInterventions === undefined
  )
  const {scenario} = useScenario(scenarioIdForInterventions ?? "", scenarioIdForInterventions === undefined)
  const {updateQuestionnaireQuestion, updateQuestionnaireQuestionLoading} = useUpdateQuestionnaireQuestion(questionId)
  const {updateQuestionnaireAnswer, updateQuestionnaireAnswerLoading} = useUpdateQuestionnaireAnswer(questionId)
  const {
    updateFreetextQuestionCodingCriterion,
    updateFreetextQuestionCodingCriterionLoading
  } = useUpdateFreetextQuestionCodingCriterion(questionId)
  const {deleteEntity: deleteQuestion, deleteEntityLoading} = useDeleteQuestionnaireQuestion(questionnaireId)
  const {deleteEntity: deleteAnswer, deleteEntityLoading: deleteAnswerLoading} = useDeleteQuestionnaireAnswer(
    questionId
  )
  const {
    deleteEntity: deleteFreetextCriterion,
    deleteEntityLoading: deleteFreetextCriterionLoading
  } = useDeleteFreetextQuestionCodingCriterion(questionId)
  const {createQuestionnaireAnswer, createQuestionnaireAnswerLoading} = useCreateQuestionnaireAnswer()
  const {
    createFreetextQuestionCodingCriterion,
    createFreetextQuestionCodingCriterionLoading
  } = useCreateFreetextQuestionCodingCriterion()
  const {repositionQuestionnaireAnswer, repositionQuestionnaireAnswerLoading} = useRepositionQuestionnaireAnswer(
    questionId
  )

  const isFreetext = questionOption.map(question => question.questionType === QuestionType.FreeText).getOrElse(false)
  const handleToggleFreeTextAnswer = (isAdditionalTextAnswerAllowed: boolean) =>
    questionOption.forEach(question => {
      updateQuestionnaireQuestion({
        isAdditionalTextAnswerAllowed,
        ...pick(question, "binaryFileId", "text", "questionType", "scoringType", "score")
      })
    })
  const handleCreateAnswer = () => {
    if (isFreetext) {
      createFreetextQuestionCodingCriterion({description: "", questionId, score: 1})
    } else {
      createQuestionnaireAnswer({questionId, isCorrect: false, text: ""})
    }
  }
  const createDefaultHolisticAnswer = () =>
    isFreetext && createFreetextQuestionCodingCriterion({description: "", questionId, score: 0})
  const handeUpdateQuestionnaireAnswer = (answerId: UUID, update: QuestionAnswerTableEntity) => {
    if (isFreetext) {
      return updateFreetextQuestionCodingCriterion(answerId, {score: update.score || 0, description: update.text})
    } else {
      // make sure single choice selection clears other selected answers
      questionOption.forEach(question => {
        if (question.questionType === QuestionType.SingleChoice && (update as QuestionnaireAnswerUpdate).isCorrect) {
          question.answers
            .filter(answer => answer.id !== answerId && answer.isCorrect)
            .forEach(answer => updateQuestionnaireAnswer(answer.id, {text: answer.text, isCorrect: false}))
        }
      })

      return updateQuestionnaireAnswer(answerId, {
        text: update.text,
        isCorrect: !!update.isCorrect
      })
    }
  }
  const repositionAnswers = (orderedEntities: ResortedEntity[]) =>
    Promise.all(orderedEntities.map(({id, predecessorId}) => repositionQuestionnaireAnswer(id, predecessorId)))
  const sortableAnswers = convertAnswersToResortableEntity(questionOption.map(data => data.answers))

  const handleDeleteAnswer = (id: UUID) => {
    if (isFreetext) {
      return deleteFreetextCriterion(id)
    } else {
      return deleteAnswer(id)
    }
  }

  const handleDeleteQuestion = (id: UUID) =>
    new Promise<void>((resolve, reject) => {
      deleteQuestion(id)
        .then(() => {
          navigateToQuestionnaire()
          resolve()
        })
        .catch(reject)
    })

  const navigateToInterventions = () =>
    dispatch(
      navigateToRouteAction(Route.ScenarioInterventionsGroupEntityDetail, {
        scenarioId: scenarioIdForInterventions,
        groupType: InterventionGroupType.Event,
        headerGroupType: InterventionHeaderGroupType.Event,
        groupEntityId: questionnaireId
      })
    )

  return {
    actionsLoading:
      updateQuestionnaireQuestionLoading ||
      deleteEntityLoading ||
      deleteAnswerLoading ||
      deleteFreetextCriterionLoading ||
      createQuestionnaireAnswerLoading ||
      updateQuestionnaireAnswerLoading ||
      repositionQuestionnaireAnswerLoading ||
      updateFreetextQuestionCodingCriterionLoading ||
      interventionsLoading ||
      createFreetextQuestionCodingCriterionLoading,
    dataLoading: questionnaireQuestionLoading,
    deleteAnswer: handleDeleteAnswer,
    deleteQuestion: handleDeleteQuestion,
    handleCreateAnswer,
    isChangeQuestionTypeModalVisible,
    interventionsConfig: {
      isCreateInterventionModalVisible,
      isScenarioReadOnly: scenario.exists(scenario => scenario.publishedAt !== null || scenario.finalizedAt !== null),
      runtimeSurveyAnswerSelectionInterventions: getInterventionsForTypeName(
        interventions.getOrElse([]),
        "RuntimeSurveyAnswerSelectionIntervention"
      ) as RuntimeSurveyAnswerSelectionIntervention[],
      setSelectedAnswerForInterventionsSetting,
      toggleIsCreateInterventionModalVisible: () =>
        setIsCreateInterventionModalVisible(!isCreateInterventionModalVisible),
      scenarioMaxDurationInSeconds: scenario.map(scenario => scenario.maxDurationInSeconds || 0).getOrElse(0),
      selectedAnswerForInterventionsSetting,
      navigateToInterventions
    },
    openAnswersSortOverlay: () => setResortModalVisible(true),
    question: questionOption,
    repositionAnswers,
    resortModalVisible,
    setChangeQuestionTypeModalVisible,
    setResortModalVisible,
    sortableAnswers,
    toggleFreeTextAnswer: handleToggleFreeTextAnswer,
    updateQuestionnaireAnswer: handeUpdateQuestionnaireAnswer,
    updateQuestionnaireQuestion,
    createDefaultHolisticAnswer
  }
}
