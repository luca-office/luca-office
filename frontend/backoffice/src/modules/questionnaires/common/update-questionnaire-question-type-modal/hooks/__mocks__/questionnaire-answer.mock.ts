import {QuestionnaireAnswer} from "shared/models"

export const questionnaireAnswerMock = (
  questionId: UUID = "18316dac-2355-4e45-a7f3-f339461a488f"
): QuestionnaireAnswer => ({
  __typename: "QuestionnaireAnswer",
  createdAt: new Date(2020, 10, 5).toISOString(),
  modifiedAt: new Date(2020, 10, 15).toISOString(),
  id: "1fce8289-71fa-4568-bccc-329dc4cc67eb",
  text: "answer",
  questionId,
  isCorrect: false,
  position: 1
})
