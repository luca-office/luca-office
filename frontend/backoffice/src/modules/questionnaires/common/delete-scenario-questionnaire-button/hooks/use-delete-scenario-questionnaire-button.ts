import {useDispatch} from "react-redux"
import {DeleteScenarioQuestionnaireProps, useDeleteScenarioQuestionnaire} from "shared/graphql/hooks"
import {useErrorHandler} from "shared/utils"
import {navigateToRouteAction} from "../../../../../redux/actions/navigation-action"
import {Route} from "../../../../../routes"

export interface UseDeleteScenarioQuestionnaireButtonProps
  extends Pick<DeleteScenarioQuestionnaireProps, "deleteScenarioQuestionnaireLoading"> {
  readonly deleteScenarioQuestionnaire: () => void
}

export const useDeleteScenarioQuestionnaireButton = (
  questionnaireId: UUID,
  scenarioId: UUID
): UseDeleteScenarioQuestionnaireButtonProps => {
  const dispatch = useDispatch()

  const handleError = useErrorHandler()

  const {deleteScenarioQuestionnaire, deleteScenarioQuestionnaireLoading} = useDeleteScenarioQuestionnaire(
    scenarioId,
    questionnaireId,
    err => handleError(err.graphQLErrors[0], "error_entity_in_use_scenario_questionnaire_intervention")
  )
  const handleDelete = () => {
    deleteScenarioQuestionnaire().then(() => {
      dispatch(navigateToRouteAction(Route.ScenarioQuestionnaireDetail, {scenarioId}))
    })
  }

  return {
    deleteScenarioQuestionnaire: handleDelete,
    deleteScenarioQuestionnaireLoading
  }
}
