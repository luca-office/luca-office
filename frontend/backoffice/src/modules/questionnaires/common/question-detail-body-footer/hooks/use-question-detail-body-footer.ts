import React from "react"
import {useScenario, useScenarioQuestionnaires, useUpdateScenarioQuestionnaire} from "shared/graphql/hooks"
import {convertSecondsToMinutes} from "shared/utils"
import {defaultEventActivationDelaySeconds} from "../../../config"

export interface UseQuestionDetailBodyFooterHook {
  readonly displayEventDelayModal: boolean
  readonly setDisplayEventDelayModal: (isVisible: boolean) => void
  readonly loading: boolean
  readonly activationDelaySeconds: number
  readonly onUpdateDelay: (activationDelaySeconds: number) => void
  readonly maxScenarioDuration: number
  readonly existingDurations: number[]
}

export const useQuestionDetailBodyFooter = (
  questionnaireId: UUID,
  scenarioId: UUID
): UseQuestionDetailBodyFooterHook => {
  const [displayEventDelayModal, setDisplayEventDelayModal] = React.useState<boolean>(false)

  const {scenario: scenarioOption, scenarioLoading} = useScenario(scenarioId)
  const {scenarioQuestionnaires, scenarioQuestionnairesLoading} = useScenarioQuestionnaires(scenarioId)
  const {updateScenarioQuestionnaire} = useUpdateScenarioQuestionnaire()

  const selectedQuestionnaire = scenarioQuestionnaires.map(questionnaires =>
    questionnaires.find(questionnaire => questionnaire.questionnaireId === questionnaireId)
  )

  const maxScenarioDuration = scenarioOption
    .map(scenario => (scenario.maxDurationInSeconds ? convertSecondsToMinutes(scenario.maxDurationInSeconds) : -1))
    .getOrElse(-1)

  const activationDelaySeconds = selectedQuestionnaire
    .map(questionnaire => questionnaire?.activationDelayInSeconds ?? defaultEventActivationDelaySeconds)
    .getOrElse(defaultEventActivationDelaySeconds)

  const onUpdateDelay = (activationDelayInSeconds: number) =>
    updateScenarioQuestionnaire(scenarioId, questionnaireId, {activationDelayInSeconds: activationDelayInSeconds})

  return {
    loading: scenarioLoading || scenarioQuestionnairesLoading,
    activationDelaySeconds,
    displayEventDelayModal,
    setDisplayEventDelayModal,
    onUpdateDelay,
    maxScenarioDuration,
    existingDurations: scenarioQuestionnaires
      .getOrElse([])
      .filter(sq => sq.questionnaireId !== questionnaireId)
      .map(sq => convertSecondsToMinutes(sq.activationDelayInSeconds))
  }
}
