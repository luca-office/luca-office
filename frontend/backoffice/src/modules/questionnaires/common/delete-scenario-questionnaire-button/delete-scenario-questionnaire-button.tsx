import * as React from "react"
import {deleteEntityButtonStyles, OrlyButtonContainer} from "shared/components"
import {useDeleteScenarioQuestionnaireButton} from "./hooks/use-delete-scenario-questionnaire-button"

interface Props {
  readonly questionnaireId: UUID
  readonly scenarioId: UUID
  readonly disabled?: boolean
}

export const DeleteScenarioQuestionnaireButton: React.FunctionComponent<Props> = ({
  questionnaireId,
  scenarioId,
  disabled
}) => {
  const {deleteScenarioQuestionnaire, deleteScenarioQuestionnaireLoading} = useDeleteScenarioQuestionnaireButton(
    questionnaireId,
    scenarioId
  )

  return (
    <OrlyButtonContainer
      onConfirm={deleteScenarioQuestionnaire}
      iconColor={"white"}
      disabled={deleteScenarioQuestionnaireLoading || disabled}
      modalTextKey={"scenario_events__header_orly_delete_text"}
      modalTitleKey={"scenario_events__header_orly_delete_title"}
      customButtonStyles={deleteEntityButtonStyles.trashButton}
    />
  )
}
