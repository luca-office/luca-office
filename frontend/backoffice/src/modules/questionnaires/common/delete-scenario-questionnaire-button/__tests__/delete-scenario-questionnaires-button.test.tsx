import {shallow} from "enzyme"
import * as React from "react"
import {create} from "react-test-renderer"
import {OrlyButtonContainer} from "shared/components"
import {questionnairesMock, scenariosMock} from "shared/graphql/__mocks__"
import {DeleteScenarioQuestionnaireButton} from "../delete-scenario-questionnaire-button"
import * as hook from "../hooks/use-delete-scenario-questionnaire-button"
import {UseDeleteScenarioQuestionnaireButtonProps} from "../hooks/use-delete-scenario-questionnaire-button"

const scenarioId = scenariosMock[0].id
const questionnaireId = questionnairesMock[0].id
const hookValuesDefault: UseDeleteScenarioQuestionnaireButtonProps = {
  deleteScenarioQuestionnaireLoading: false,
  deleteScenarioQuestionnaire: jest.fn
}

const hookStateSpy = jest.spyOn(hook, "useDeleteScenarioQuestionnaireButton")

describe("scenario-detail", () => {
  it("renders correctly", () => {
    hookStateSpy.mockReturnValue(hookValuesDefault)
    const component = create(<DeleteScenarioQuestionnaireButton {...{scenarioId, questionnaireId}} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it("has correct structure", () => {
    hookStateSpy.mockReturnValue(hookValuesDefault)
    const tree = shallow(<DeleteScenarioQuestionnaireButton {...{scenarioId, questionnaireId}} />)

    expect(tree.find(OrlyButtonContainer)).toHaveLength(1)
  })
})
