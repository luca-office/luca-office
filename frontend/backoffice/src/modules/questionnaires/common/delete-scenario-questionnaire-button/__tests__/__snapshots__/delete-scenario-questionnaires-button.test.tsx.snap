// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`scenario-detail renders correctly 1`] = `
<div
  className="css-0"
>
  <button
    className="css-q415zv-Button"
    disabled={false}
    onClick={[Function]}
  >
    <div
      className="css-95j2hm-Button"
    >
      <div
        className="css-b32040-Icon"
        dangerouslySetInnerHTML={
          {
            "__html": {},
          }
        }
      />
    </div>
  </button>
</div>
`;
